pkg/apiserver/resthandler.go

```diff
@@ -82,7 +82,13 @@ func GetResource(r RESTGetter, ctxFn ContextFunc, namer ScopeNamer, codec runtim
 }
 
 func parseSelectorQueryParams(query url.Values, version, apiResource string) (label, field labels.Selector, err error) {
-	label, err = labels.Parse(query.Get("labels"))
+	var labelString string
+	if version == "v1beta1" || version == "v1beta2" {
+		labelString = query.Get("labels")
+	} else {
+		labelString = query.Get("label-selector")
```

Comment:
This logic is right now spread at 2 places: here and in pkg/client/request.go.
How about adding a function that takes the apiVersion and returns the appropriate QueryParamName?
We can then use the same function while documenting the query param with go-restful.